djangocms_link/models.py

```diff
@@ -145,6 +145,66 @@ def get_link(self):
 
         return link
 
+    def clean(self):
+        super(AbstractLink, self).clean()
+        field_names = (
+            'external_link',
+            'internal_link',
+            'mailto',
+            'phone',
+        )
+        anchor_field_name = 'anchor'
+        field_names_allowed_with_anchor = (
+            'external_link',
+            'internal_link',
+        )
+
+        cleaned_data = {
```

Comment:
this is not needed as you already create a dict for link_fields.
Instead you can do:

``anchor_field_value = getattr(self, anchor_field_name)``